test-examples.sh

```
@@ -0,0 +1,350 @@
+    fi
+    
+    # Run the generated script from the example directory
+    echo "  Running code generation..."
+    pushd "$example_dir" >/dev/null
+    local gen_output
+    local gen_exit_code
+    gen_output=$("$output_path/bin/bufrnix" 2>&1)
+    gen_exit_code=$?
+    
+    # Filter out known warnings
+    local filtered_output=$(echo "$gen_output" | grep -v "warning: directory does not exist" | grep -v "^$")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)